cheri/cheri_insts.sail

```diff
@@ -1036,6 +1038,51 @@ function clause execute (CLoad(rd, cb, rt, offset, signext, width, linked)) =
     }
 }
 
+union clause ast = CLoadTags : (regno, regno)
+function clause execute (CLoadTags(rd, cb)) =
+{
+  checkCP2usable();
+  cb_val = readCapRegDDC(cb);
+  if not (cb_val.tag) then
+    raise_c2_exception(CapEx_TagViolation, cb)
+  else if (cb_val.sealed) then
+    raise_c2_exception(CapEx_SealViolation, cb)
+  else if not (cb_val.permit_load) then
+    raise_c2_exception(CapEx_PermitLoadViolation, cb)
+  else
+    {
+      let vAddr   = getCapCursor(cb_val) % pow2(64);
+      let vAddr64 = to_bits(64, vAddr);
+      if ((vAddr + 64) > getCapTop(cb_val)) then
+        raise_c2_exception(CapEx_LengthViolation, cb)
+      else if (vAddr < getCapBase(cb_val)) then
+        raise_c2_exception(CapEx_LengthViolation, cb)
+      else if not (vAddr64[5..0] == 0b000000) then /* XXX Cache line width? */
+        SignalExceptionBadAddr(AdEL, vAddr64)
+      else
+        {
+          pAddr      = (TLBTranslate(vAddr64, LoadData));
+
+          /* XXX This cannot possibly be idiomatic sail */
```

Comment:
I think you should be able to use infix operator @ for vector concat and might get away with an implicit cast from bool to bit(1). Some type annotations might be required to help.